docs/tutorials/using-backstage-proxy-within-plugin.md

```diff
@@ -0,0 +1,210 @@
+---
+id: using-backstage-proxy-within-plugin
+title: Using the Backstage Proxy from Within a Plugin
+# prettier-ignore
+description: Guide on how to create a set of API bindings that interface with a backend via the backstage proxy
+---
+
+This guide walks you through setting up a simple proxy to an existing API that
+is deployed externally to backstage and sending requests to that API from within
+a backstage frontend plugin. For a more detailed description of the APIs used
+here, check out the [Utility APIs section](../api/utility-apis.md).
+
+If your plugin requires access to an API, backstage offers
+[3 options](../plugins/call-existing-api.md):
+
+1. you can
+   [access the API directly](../plugins/call-existing-api.md#issuing-requests-directly),
+1. you can create a [backend plugin](../plugins/backend-plugin.md) if you are
+   implementing the API alongside your frontend plugin
+1. you can configure backstage to proxy to an already existing API. The third
+   approach tends to be the most popular, since you likely already have an
+   existing API, and it allows you to develop/deploy your API in any way you
+   want.
+
+**Table of Contents**
+
+- [Setting up the backstage proxy](#setting-up-the-backstage-proxy)
+- [Calling an API using the backstage proxy](#calling-an-api-using-the-backstage-proxy)
+  - [Defining the API client interface](#defining-the-api-client-interface)
+  - [Creating the API client](#creating-the-api-client)
+  - [Bundling your ApiRef with your plugin](#bundling-your-apiref-with-your-plugin)
+  - [Using your plugin in your components](#using-your-plugin-in-your-components)
+
+# Setting up the backstage proxy
+
+Let's say your plugin's API is hosted at _https://api.myawesomeservice.com/v1_,
+and you want to be able to access it within backstage at
+`/api/proxy/<your-proxy-uri>`, and add a default header called
+`X-Custom-Source`. You will need to add the following to `app-config.yaml`:
+
+```yaml
+proxy:
+  '/<your-proxy-uri>':
+    target: https://api.myawesomeservice.com/v1
+    changeOrigin: true # Use this to avoid cors related issues
```

Comment:
Perhaps skip this one or comment it out, just to not promote the usage of it by default